### PR TITLE
Vote encoding - part 6 (Maybe final)

### PR DESCRIPTION
**Pull request by av-michael**
This PR includes:
- Changes to av clients constructBallot signature (now accepts new BallotSelection instead of CastVoteRecord)
- Changes to av verifier decryptBallot signature (now returns an array of ContestSelection)
- New method in av verifier getReadableContestSelections
- Mocks have been rerecorded using branches in dbb and avx not yet on main
- Tests have been adapted to use new ballot selection structure

### CODE REVIEW

File: test/construct_ballot.test.ts [hunk 1]
```diff
@@ -0,0 +1,153 @@
+import { AVClient } from '../lib/av_client';
```
Comment (av-mads) [inline test/construct_ballot.test.ts:1]: What are we actually testing?

**Review by av-mads — APPROVED** [review]

File: lib/av_client/construct_contest_envelopes.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+function extractConfig( state: ClientState ){
+  if( !state.voterSession ){
+    throw new InvalidStateError('Cannot construct ballot cryptograms. Voter registration not completed successfully')
+  }
+
+  if( !state.electionConfig ){
+    throw new InvalidStateError('Cannot construct ballot cryptograms. No election configuration present')
+  }
```
[the remark this answers is not in the document]
Comment (av-michael) [reply lib/av_client/construct_contest_envelopes.ts]: Yes

File: lib/av_client.ts [hunk 3]
```diff
@@ -467,6 +466,20 @@ export class AVClient implements IAVClient {
     return this.electionConfig
   }
 
+  public getVoterSession(): VoterSessionItem {
+    if(!this.electionConfig){
+      throw new InvalidStateError('No voter session loaded')
```
[the remark this answers is not in the document]
Comment (av-michael) [reply lib/av_client.ts:470]: Fixed

File: test/construct_ballot.test.ts (hunk 1 above)
Comment (av-michael) [reply test/construct_ballot.test.ts:1]: @lukelex, @av-emil and I have discussed the problem that you highlight here, and we all agree that it is wrong to use mocks from another test. Each test should have independant mocks. But for now we leave it as is. Any fixes to this should be addressed in a later pull request

File: lib/av_client/construct_contest_envelopes.ts [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+}
+
+type ConstructResult = {
+  pedersenCommitment: {
+    commitment: string,
+    randomizer: string
+  }
+  envelopeRandomizers: ContestMap<string[]>
+  contestEnvelopes: ContestEnvelope[],
+}
```
[the remark this answers is not in the document]
Comment (av-michael) [reply lib/av_client/construct_contest_envelopes.ts:46]: Personally I do not think so. Defining a return type locally enables the type checker to throw errors on build time, if the returned values does not include the required types. It does its thing, but should not pollute the global types.

**Pull request merged**